### DISCUSSION TITLE
[Android] Implement vertical center for Label

### DISCUSSION THREAD

**Discussion by kefahB** [NativeScript/NativeScript#10058] [Ideas]
Hi,

It will be cool to implement `Gravity`with `verticalAlignment="center"` for Label on android, for the moment we need to implement lloaded event on each label and declare the flowing for each page!

```ts
export function lblLoaded(args: EventData) {
	let lbl = <View>args.object;
    if(isAndroid) {
        lbl.android.setGravity(17)
    }
}
```

**Comment by farfromrefug** [comment]
@kefahB alreadly working in `@nativescript-community/ui-label` with `verticalTextAlignment`. At one point we might be using it in the core too

**Comment by NathanWalker** [comment]
I think this should be the default in core - iOS defaults to center and Android should be the same. I’ve long been frustrated about this, thanks for bringing this up @kefahB. If no one has good reasons they should remain different let’s just make this the default in 8.4.

I use ui-label all the time as well and I think it would be great for that to replace the core implementation @farfromrefug - using transient or implement directly in core?

**Reply by farfromrefug** [reply]
@NathanWalker the one thing i am not sure about in making it the default is that the iOS implementation uses `UITextView` instead of `UILabel`. Not sure if we care or not ...

**Reply by kefahB** [reply]
I think it is a good think when we see the same behavior on IOS and android as we develop for both platforms, it cost as less work

**Reply by NathanWalker** [reply]
https://github.com/NativeScript/NativeScript/pull/10233

**Comment by kefahB** [comment]
@NathanWalker Did you intent to push this feature into the 8.0.4 ?

**Reply by NathanWalker** [reply]
It will be on by default with 8.5.

**Reply by NathanWalker** [reply]
@kefahB https://github.com/NativeScript/NativeScript/pull/10233

**Reply by kefahB** [reply]
@NathanWalker Grate thank you 🙌

**Comment by vallemar** [comment]
I don't know if this would be nice by default or not, but this feature would definitely be nice.  On many occasions you have to put the label in a layout to center it.  It would be nice to be able to center it to remove views from layouts

**Comment by NathanWalker** [comment]
This will be included in 8.5 release: https://github.com/NativeScript/NativeScript/pull/10233
